crates/sui-graphql-rpc/schema.graphql

```diff
@@ -0,0 +1,840 @@
+  gasPrice: U64
+  commissionRate: U64
+  nextEpochStake: U64
+  nextEpochGasPrice: U64
+  nextEpochCommissionRate: U64
+
+  # The number of epochs for which this validator has been below the
+  # low stake threshold.
+  atRisk: Int
+
+  # The other validators this validator has reported
+  reportRecords: [SuiAddress]
+
+  apy: Float
+}
+
+type ValidatorCredentials {
+  protocolPubKey: Bytes
+  networkPubKey: Bytes
+  workerPubKey: Bytes
+  proofOfPossession: Bytes
+
+  netAddress: String
+  p2pAddreess: String
+  primaryAddress: String
+  workerAddress: String
+}
+
+type StorageFund {
+  totalObjectStorageRebates: U64
+  nonRefundableBalance: U64
+}
+
+type SafeMode {
+  enabled: Boolean
+  gasSummary: GasCostSummary
+}
+
+type Checkpoint {
+  digest: Digest!
+  sequenceNumber: U64!
+  timestampMs: U64
+  validatorSignature: Bytes
+
+  # Commitments
+  previousCheckpoint: Digest
+  liveObjectSet: Digest
+
+  networkTotalTransactions: U64
+  rollingGasSummary: GasCostSummary
+
+  epoch: Epoch
+
+  # Only available on the final checkpoint in an epoch
+  endOfEpoch: EndOfEpochData
+
+  transactions: TransactionBlockConnection
+}
+
+type EndOfEpochData {
+  newCommittee: [CommitteeMember]
+  nextProtocolVersion: Int
+}
+
+type CommitteeMember {
+  authorityName: String
+  stakeUnit: Int
+}
+
+type TransactionBlock {
+  digest: Digest!
+  sender: Address
+  gasInput: GasInput
+  kind: TransactionBlockKind
+  signatures: [TransactionSignature]
+  effects: TransactionBlockEffects
+
+  expiration: Epoch
+
+  bcs: Bytes
+}
+
+enum TransactionBlockKindInput {
+  CONSENSUS_COMMIT_PROLOGUE
+  GENESIS
+  CHANGE_EPOCH
+  PROGRAMMABLE
+}
+
+union TransactionBlockKind =
+    ConsensusCommitPrologueTransaction
+  | GenesisTransaction
+  | ChangeEpochTransaction
+  | ProgrammableTransactionBlock
+
+type ConsensusCommitPrologueTransaction {
+  epoch: Epoch
+  # TODO: This is the "leader round" -- does this line up with
+  # checkpoints? In which case, it may suffice to have a `Checkpoint`
+  # here.
+  round: U64
+  timestampMs: U64
+}
+
+type GenesisTransaction {
+  objects: [Object]
+}
+
+type ChangeEpochTransaction {
+  epoch: Epoch
+  timestampMs: U64
+
+  storageCharge: U64
+  computationCharge: U64
+  storageRebate: U64
+}
+
+type ProgrammableTransactionBlock {
+  inputs: [TransactionInput]
+  transactions: [ProgrammableTransaction]
+}
+
+union TransactionInput = MovePackage | MoveObject | ShareInput
+
+type SharedInput {
+  id: SuiAddress
+  initialSharedVersion: U64
+  mutable: Boolean
+}
+
+union TransactionArgument = GasCoin | Input | Result
+
+type GasCoin {}
+type Input   { ix: Int }
+type Result  { cmd: Int, ix: Int }
+
+union ProgrammableTransaction =
+    MoveCallTransaction
+  | TransferObjectsTransaction
+  | SplitCoinTransaction
+  | MergeCoinsTransaction
+  | PublishTransaction
+  | UpgradeTransaction
+  | MakeMoveVecTransaction
+
+type MoveCallTransaction {
+  moduleId: MoveModuleId
+  name: String
+  function: NormalizedFunction
```

Comment:
This could technically just be `function: NormalizedFunction`, thoughts?

Reply:
I dont understand this comment. The field _is_ `function: NormalizedFunction`

Reply:
Or are you saying to omit the `moduleId` and `name` fields since `NormalizedFunction` already contains them?
In that case yes, omit.